packages/kbn-content-management-utils/src/saved_object_content_storage.ts

```diff
@@ -142,6 +157,11 @@ export interface SOContentStorageConstructorParams<Types extends CMCrudTypes> {
   createArgsToSoCreateOptions?: CreateArgsToSoCreateOptions<Types>;
   updateArgsToSoUpdateOptions?: UpdateArgsToSoUpdateOptions<Types>;
   searchArgsToSOFindOptions?: SearchArgsToSOFindOptions<Types>;
+  itemAttrsToSavedObjectAttrs?: ItemAttrsToSavedObjectAttrs<
+    Types['ContentAttributes'],
+    Types['Attributes']
+  >;
+  savedObjectToItem?: SavedObjectToItem<Types['Attributes']>;
```

Comment:
Two optional methods are added to the abstract SOContentStorage class that specify in/out transform functions for attributes between the persisted saved object and the content management output. This means we do not need any saved object migrations. If transform methods are not specified the saved object attributes are returned.